### ISSUE TITLE
Begin to write xlsx export function for GRNsight

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#683]
Once @jtorre39 wraps up his work with GRNmap, it will be time to switch over to writing the new functionality for GRNsight.  The first task will be to take the existing data stored by GRNsight's json format and export it into .xlsx format (for GRNmap).  I believe when @eileenchoe wrote the new data structure to capture expression data, she actually captured all information from the GRNmap input and output workbooks.  So, @jtorre39 can look to see if that is true.  If so, then he should be able to input a GRNmap input or output workbook into GRNsight and then export it back to xlsx and get an identical file.

**Comment by kdahlquist** [comment]
@jtorre39 has finished his work with GRNmap as the release of v1.10 is immanent.  He is now starting to investigate this issue.

**Comment by jtorre39** [comment]
@dondi question came up, is there an expected behavior for producing the `optimization_parameters` sheet for graphml or sif imported files? 

From what I've seen, if it's a spreadsheet we're importing we do store the it in a meta property within the network object. But the sif and graphml importers don't seem to handle any logic for populating something I can use to export to the `optimization_parameters` sheet.

**Comment by jtorre39** [comment]
After examining the network object when importing GraphML and SIF, the meta property which normally populates the `optimization_parameters` sheet is empty. Unless I'm missing something here, we might have to come up with a dialogue box to populate the sheet or come up with default settings to put into the sheet.

**Comment by kdahlquist** [comment]
You are right, sif and graphml imports are only going to have the network, not any of the other sheets. Later, if when we have populated our expression data database, we will get fancy and have a dialog to have the user populate those sheets. In the meantime, we can give a message saying that they are only going to get the network or network optimized weights sheet.

Also, unrelated, you should meet with @mihirsamdarshi to do a code walkthrough.

Kam D. Dahlquist, Ph.D.
Professor of Biology
Loyola Marymount University

-------- Original message --------

**Comment by kdahlquist** [comment]
@jtorre39 was working on GRNmap last week, but has also figured out the data structure for GRNsight.  There are some more bugs with GRNmap that need to be tracked down, so this will be on hold a little until those are addressed.

**Comment by jtorre39** [comment]
Spent most of the work session debugging the GRNmap issues that came up. Work on those are mostly done. Will spend a little under an hour sometime before the upcoming meeting working with @jiajialingling so we can begin writing the tests for export.

**Comment by jtorre39** [comment]
Worked with @jiajialingling to ramp her up on some JS, Git, and GRNsight stuff. Had her push her first GRNsight commit as we began to make tests for xlsx-export.

**Comment by jtorre39** [comment]
Grinded out a little more of the test for GRNsight. Hopefully GRNmap release is taken care of so I can fully focus on the working on this. I should be able to finish the test and begin implementation work the upcoming session.

**Comment by jtorre39** [comment]
Finished setting up the test last work session, and began implementation work. I should make substantial progress on implementation work next work session after the weekly meeting!

**Comment by jtorre39** [comment]
My update for this week is I am continuing to crank away at implementation work internally. I spoke to @dondi about a reasonable finish line for my work this semester being have the function for this implemented and fully tested internally but not hooked up to any front-end components yet. 

I am aiming for a PR for implementation and test by next meeting.

**Comment by jtorre39** [comment]
Got the PR up! Implementation (and tests) on only the back-end side to reconvert the network data structure into an XLSX-ready object is included here.

Once this PR is merged functionality will be present for future work to hook up the front-end to this, and write out the XLSX object using node-xlsx.

**Comment by kdahlquist** [comment]
This issue is complete and superseded by #714.
